services/QuillLMS/app/controllers/cms/districts_controller.rb

```diff
@@ -28,6 +28,22 @@ def show
     @admins = DistrictAdmin.includes(:user).where(district_id: params[:id].to_i)
   end
 
+  def new
+    @district = District.new
+    @editable_attributes = editable_district_attributes
+  end
+
+  def create
```

Comment:
Before getting into `create` and `update`, I'd like to first recommend renaming: 
`def edit_or_add_district_params` to just `def district_params`.  It's a Rails convention to name the params for use in create/update as just the name of the model followed by underscore params.  Note that this is in contrast to the query params definition elsewhere. 

https://github.com/empirical-org/Empirical-Core/blob/a0604d267a9307955f6e0b8fa434c315b7f05ffd/services/QuillLMS/app/controllers/cms/districts_controller.rb#L121

In this particular case, the exception handling isn't being use to notify anything outside of the control (e.g. ErrorNotifier), so I would opt to use a conditional with  boolean `save` over rescue block with `save!` since that adds in unnecessary overhead:

```
district =  District.new(district_params)

if district.save
  redirect_to cms_district_path(district)
else
  redirect_to cms_districts_path, error: district.errors
end
```

Also note that instead you can [assign](https://edgeapi.rubyonrails.org/classes/ActionController/Redirecting.html#method-i-redirect_to) flash messages via `redirect_to`